sys/sysent.c

```diff
@@ -32,38 +32,7 @@ int sys_sbrk(thread_t *td, syscall_args_t *args) {
 
   assert(td->td_uspace);
 
-  if (!td->td_uspace->sbrk_entry) {
-    /* sbrk was not used before by this thread. */
-    size_t size = roundup(increment, PAGESIZE);
-    vm_addr_t addr;
-    if (vm_map_findspace(td->td_uspace, BRK_SEARCH_START, size, &addr) != 0)
-      return -ENOMEM;
-    vm_map_entry_t *entry = vm_map_add_entry(td->td_uspace, addr, addr + size,
-                                             VM_PROT_READ | VM_PROT_WRITE);
-    entry->object = default_pager->pgr_alloc();
-
-    td->td_uspace->sbrk_entry = entry;
-    td->td_uspace->sbrk_end = addr + increment;
-
-    return addr;
-  } else {
-    /* There already is a brk segment in user space map. */
-    vm_map_entry_t *brk_entry = td->td_uspace->sbrk_entry;
-    vm_addr_t brk = td->td_uspace->sbrk_end;
-    if (brk + increment == brk_entry->end) {
-      /* No need to resize the segment. */
-      td->td_uspace->sbrk_end = brk + increment;
-      return brk;
-    }
-    /* Shrink or expand the vm_map_entry */
-    vm_addr_t new_end = roundup(brk + increment, PAGESIZE);
-    if (vm_map_resize(td->td_uspace, brk_entry, new_end) != 0) {
-      /* Map entry expansion failed. */
-      return -ENOMEM;
-    }
-    td->td_uspace->sbrk_end += increment;
-    return brk;
-  }
+  return vm_map_sbrk(td->td_uspace, increment);
```

Comment:
I really dislike moving `sbrk` syscall handler code to `vm_map` module. This looks like a kludge. Can we do better?